errors.go

```diff
@@ -0,0 +1,16 @@
+package yoti
+
+// TemporaryError indicates that a temporary outage has occured and the
```

Comment:
I have just realised that this is not a good approach to marking temporary errors.

This is because I hadn't realised that you can use As to check for interfaces as well as concrete error structs. The std go packages already use an interface with method `Temporary() bool` that they use to indicate temporary errors. We should follow their approach and add a `Temporary` method to this error that always returns true.

This is an example of using `As` to check for an interface - https://play.golang.org/p/IVjEsfflbNJ

This means we can stop wrapping errors from `doRequest` since go will have already marked them as `Temporary` but we can still use this struct to wrap http errors for 500 status codes.

The documentation will need to be updated to show checking for `Temporary` errors and we should remove reference to the yoti.TemoraryError.